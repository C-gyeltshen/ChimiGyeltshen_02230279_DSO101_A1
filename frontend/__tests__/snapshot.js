import { render } from '@testing-library/react'
import ToDo from '../src/app/todo/page' // <-- corrected path

test('renders the ToDo component correctly', () => {
  const { container } = render(<ToDo />)
  expect(container).toMatchSnapshot()
})