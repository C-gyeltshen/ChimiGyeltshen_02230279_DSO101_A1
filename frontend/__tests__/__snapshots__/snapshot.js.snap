// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ToDo component correctly 1`] = `
<div>
  <div
    class="min-h-screen bg-gradient-to-r from-blue-50 to-indigo-50"
  >
    <div
      class="w-full flex justify-center"
    >
      <div
        class="w-5/6 h-32 bg-green-700 flex items-center justify-center rounded-b-2xl shadow-md"
      >
        <h1
          class="text-white text-4xl md:text-6xl font-bold"
        >
          Todo App
        </h1>
      </div>
    </div>
    <div
      class="container mx-auto px-4 py-8"
    >
      <div
        class="flex justify-between items-center mb-8"
      >
        <h1
          class="text-3xl font-bold text-gray-800"
        >
          My Tasks
        </h1>
        <button
          class="bg-indigo-600 hover:bg-indigo-700 text-white font-semibold py-2 px-4 rounded-lg shadow-md transition duration-300 flex items-center"
        >
          <svg
            class="h-5 w-5 mr-2"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
              fill-rule="evenodd"
            />
          </svg>
          Add Task
        </button>
      </div>
      <div
        class="bg-red-50 border border-red-200 text-red-700 px-4 py-3 rounded-lg mb-6"
      >
        Failed to load tasks. Please try again later.
      </div>
      <div
        class="bg-white rounded-xl shadow-lg p-6 mb-8"
      >
        <div
          class="flex flex-col items-center justify-center py-12"
        >
          <svg
            class="h-16 w-16 text-gray-300 mb-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
          <p
            class="text-gray-500 mb-6"
          >
            No tasks yet. Create your first task!
          </p>
          <button
            class="bg-indigo-500 hover:bg-indigo-600 text-white font-medium py-2 px-6 rounded-lg transition duration-300"
          >
            Create Task
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
